tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/AzureFunctionsTests.cs

```diff
@@ -229,6 +250,80 @@ public async Task SubmitsTraces()
                 using var s = new AssertionScope();
 
                 await AssertIsolatedSpans(spans);
+
+                spans.Count.Should().Be(expectedSpanCount);
+            }
+        }
+    }
+
+    [UsesVerify]
+    [Collection(nameof(AzureFunctionsTestsCollection))]
+    public class IsolatedRuntimeV4AspNetCore : AzureFunctionsTests
+    {
+        public IsolatedRuntimeV4AspNetCore(ITestOutputHelper output)
+            : base("AzureFunctions.V4Isolated.AspNetCore", output)
+        {
+            SetEnvironmentVariable("FUNCTIONS_WORKER_RUNTIME", "dotnet-isolated");
+        }
+
+        [SkippableFact]
+        [Trait("Category", "EndToEnd")]
+        [Trait("Category", "AzureFunctions")]
+        [Trait("RunOnWindows", "True")]
+        public async Task SubmitsTraces()
+        {
+            using var agent = EnvironmentHelper.GetMockAgent(useTelemetry: true);
+            using (await RunAzureFunctionAndWaitForExit(agent, expectedExitCode: -1))
+            {
+                const int expectedSpanCount = 26;
+                var spans = agent.WaitForSpans(expectedSpanCount);
+
+                // There are _additional_ spans created for these compared to the non-AspNetCore version
+                // These are http-client-handler-type: System.Net.Http.SocketsHttpHandler that come in around
+                // the same time as some of the `azure-functions.invoke` spans
+                // because of this they cause a lot of flake in the snapshots where they shift places
+                // opting to just scrub them from the snapshots - we also don't think that the spans provide much
+                // value so they may be removed from being traced.
+                var filteredSpans = FilterOutSocketsHttpHandler(spans);
+
+                using var s = new AssertionScope();
+
+                await AssertIsolatedSpans(filteredSpans.ToImmutableList(), $"{nameof(AzureFunctionsTests)}.Isolated.V4.AspNetCore");
+
+                spans.Count.Should().Be(expectedSpanCount);
+            }
+        }
+    }
+
+    [UsesVerify]
+    [Collection(nameof(AzureFunctionsTestsCollection))]
+    public class IsolatedRuntimeV4AspNetCoreV1 : AzureFunctionsTests
```

Comment:
We're currently not building for .NET 9, so we'll need to add `#if` around these tests if that's correct - otherwise we should add .NET 9 to the sample target tfms

Reply:
Yep thanks, put it around the `#if` for 6-8, 9 isn't supported by the NuGet